storage-bigtable/src/lib.rs

```diff
@@ -547,6 +547,67 @@ impl LedgerStorage {
         Ok(transaction_info.into())
     }
 
+    // Fetches and gets a vector of confirmed transactions via a multirow fetch
+    pub async fn get_confirmed_transactions(
+        &self,
+        signatures: &[Signature],
+    ) -> Result<Vec<ConfirmedTransactionWithStatusMeta>> {
+        debug!(
+            "LedgerStorage::get_confirmed_transactions request received: {:?}",
+            signatures
+        );
+        inc_new_counter_debug!("storage-bigtable-query", 1);
+        let mut bigtable = self.connection.client();
+
+        // Fetch transactions info
+        let keys = signatures.iter().map(|s| s.to_string()).collect::<Vec<_>>();
+        let cells = bigtable
+            .get_bincode_cells::<TransactionInfo>("tx", &keys)
+            .await?;
+
+        // Collect by slot
+        let mut slots: HashMap<Slot, HashMap<u32, String>> = Default::default();
+        for cell in cells {
+            if let (signature, Ok(TransactionInfo { slot, index, .. })) = cell {
+                slots.entry(slot).or_default().insert(index, signature);
+            }
+        }
+
+        // Fetch blocks and extract transactions
+        let keys = slots.keys().copied().collect::<Vec<_>>();
+        let data = self
+            .get_confirmed_blocks_with_data(&keys)
+            .await?
+            .zip(std::iter::repeat(&slots))
+            .filter_map(move |((slot, block), slots)| {
+                slots.get(&slot).map(move |block_txs| {
+                    let block_time = block.block_time;
+                    block.transactions.into_iter().enumerate().filter_map(
```

Comment:
I suppose it depends somewhat on how many signatures from the same block a caller includes at once, but I think it will generally be a lot faster to just index into the transactions vec using the transaction indexes from `slots` than to iterate through all the transactions regardless. Especially as blocks get bigger...

Reply:
Indeed, will change :+1:

Reply:
Ah nuts. Is it worth benching that to be sure? I'm trying to think of a better alternative.

Reply:
Can you confirm, will `get_confirmed_blocks_with_data` always return block data in the order of keys inputted?
If so, I wonder if we could zip `slots` with the `get_confirmed_blocks_with_data` results

Reply:
I guess it might have skips if any of the BT serialized blocks are misformed

Reply:
The resulting order of `get_confirmed_blocks_with_data` looks the same as the input but missed blocks not marked anyhow so we can not `zip` it with `slots`